MoreLinq.Test/FallbackIfEmptyTest.cs

```diff
@@ -39,6 +39,28 @@ public void FallbackIfEmptyWithEmptySequence()
         }
 
         [Test]
+        public void FallbackIfEmptyPreservesSourceCollectionIfPossible()
+        {
+            var source = new int[] { 1 };
+            // ReSharper disable PossibleMultipleEnumeration
+            Assert.AreSame(source.FallbackIfEmpty(12), source);
```

Comment:
Duh! My bad. All the overloading made it all very opaque at the call site. I guess in that case we can revert ccca164c20d889716ad54e22b75275e6beb24c4b and ignore having test cases.